src/App.jsx

```diff
@@ -14,33 +17,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALPHABETICALLY = 'alphabetically';
+const SORT_FIELD_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const getPreparedGoods = (goods, { sortField }) => {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField) {
+    switch (sortField) {
+      case SORT_FIELD_ALPHABETICALLY: {
+        return preparedGoods.toSorted();
+      }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+      case SORT_FIELD_LENGTH: {
+        return preparedGoods.toSorted(
+          (good1, good2) => good1.length - good2.length,
+        );
+      }
+
+      default:
+        return 0;
+    }
+  }
+
+  return preparedGoods;
+};
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverseOption, setReverseOption] = useState(false);
+  const sortedGoods = getPreparedGoods(goodsFromServer, { sortField });
+
+  const reverse = () => {
+    setReverseOption(optionList => !optionList);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const reset = () => {
+    setReverseOption(false);
+    setSortField('');
+  };
+
+  if (reverseOption) {
+    sortedGoods.reverse();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_FIELD_ALPHABETICALLY)}
+          className={cn(
+            { 'is-light': sortField !== SORT_FIELD_ALPHABETICALLY },
+            'button is-info',
+          )}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+          className={cn(
+            { 'is-light': sortField !== SORT_FIELD_LENGTH },
+            'button is-success',
+          )}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          onClick={() => reverse()}
+          className={cn({ 'is-light': !reverseOption }, 'button is-warning')}
+        >
+          Reverse
+        </button>
+
+        <button
```

Comment:
Consider showing this button conditionally (if the reversed is settled or if sorting is setted)